manager/base.py

```diff
@@ -242,16 +242,11 @@ def vmaas_call(cls, endpoint, data):
         headers = {'Content-type': 'application/json',
                    'Accept': 'application/json'}
         try:
-            response = requests.post(VMAAS_HOST + endpoint,
-                                     data=json.dumps(data), headers=headers)
-        except requests.exceptions.ConnectionError:
-            LOGGER.error('Could not connect to %s', (VMAAS_HOST,))
-            raise ApplicationException('Could not connect to %s' % (VMAAS_HOST,), 500)
-        if response.status_code == 200:
-            return response.json()
-        LOGGER.error('Received %s from vmaas on %s endpoint', response.status_code, endpoint)
-        raise ApplicationException('Received %s from vmaas on %s endpoint' %
-                                   (response.status_code, VMAAS_HOST + endpoint), response.status_code)
+            return vmaas_post_request(VMAAS_HOST + endpoint,
+                                      data_json=data, headers=headers)
+        except Exception:  # pylint: disable=broad-except
```

Comment:
I think it's enough to check only if response from vmaas_post_request function is not None

if it's None then raise ApplicationException() with 500 code but message should be more general than "Could not connect" - there could be other reasons why it failed, for example "Unable to fetch data from" is better